File: app/schemas/post.py
from datetime import date, datetime
from pydantic import BaseModel


class PostOut(BaseModel):
    id: str
    user: dict
    media_type: str
    preview_url: str | None = None
    thumb_url: str | None = None
    caption: str | None = None
    country: str | None = None
    city: str | None = None
    likes_count: int = 0
    comments_count: int = 0
    views_count: int = 0
    challenge_date: date | None = None
    created_at: datetime | None = None
    is_liked: bool = False
    is_mine: bool = False
    can_delete: bool = False

    model_config = {"from_attributes": True}


class PostFeedItem(BaseModel):
    id: str
    user: dict
    media_type: str
    preview_url: str | None = None
    thumb_url: str | None = None
    caption: str | None = None
    country: str | None = None
    city: str | None = None
    likes_count: int = 0
    comments_count: int = 0
    views_count: int = 0
    challenge_date: date | None = None
    created_at: datetime | None = None
    is_liked: bool = False
    is_mine: bool = False
    can_delete: bool = False


class FeedResponse(BaseModel):
    items: list[PostFeedItem]
    next_cursor: str | None = None


class BestMomentResponse(BaseModel):
    post: PostFeedItem | None = None


class CreatePostResponse(BaseModel):
    id: str
    status: str

File: android/app/src/main/java/com/bghitech/momenta/domain/model/Post.kt
package com.bghitech.momenta.domain.model

data class Post(
    val id: String,
    val user: User,
    val mediaType: String,
    val previewUrl: String,
    val thumbUrl: String?,
    val caption: String?,
    val country: String?,
    val city: String?,
    val likesCount: Int,
    val commentsCount: Int,
    val viewsCount: Int,
    val challengeDate: String,
    val createdAt: String,
    val isLiked: Boolean = false,
    val isBookmarked: Boolean = false,
    val bookmarkedAt: String? = null,
    val isMine: Boolean = false,
    val canDelete: Boolean = false
)
